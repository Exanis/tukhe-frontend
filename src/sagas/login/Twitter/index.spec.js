import TwitterLoginSaga, * as saga from './index';

window.open = jest.fn();

it('Should execute steps as expected when there is no token', () => {
    const gen = saga.twitterLoginSaga();

    jest.spyOn(saga, 'pollTwitterPopup');
    saga.pollTwitterPopup.mockImplementation(() => {
        return false;
    });

    const twitterFetch = gen.next();

    expect(twitterFetch).toMatchSnapshot();
    expect(twitterFetch.value.payload.fn()).toMatchSnapshot();
    expect(gen.next({
        data: {
            oauth_token: 'test'
        }
    })).toMatchSnapshot();

    const finalizePreparation = gen.next();

    expect(finalizePreparation).toMatchSnapshot();
    expect(finalizePreparation.value.payload.fn()).toMatchSnapshot();
    expect(gen.next({data: {}}).done).toBeTruthy();
});

it('Should execute steps as expected when there is a token', () => {
    const gen = saga.twitterLoginSaga();

    jest.spyOn(saga, 'pollTwitterPopup');
    saga.pollTwitterPopup.mockImplementation(() => {
        return false;
    });

    const twitterFetch = gen.next();

    expect(twitterFetch).toMatchSnapshot();
    expect(twitterFetch.value.payload.fn()).toMatchSnapshot();
    expect(gen.next({
        data: {
            oauth_token: 'test'
        }
    })).toMatchSnapshot();

    const finalizePreparation = gen.next();

    expect(finalizePreparation).toMatchSnapshot();
    expect(finalizePreparation.value.payload.fn()).toMatchSnapshot();
    expect(gen.next({data: {
        token: 'test'
    }})).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should handle error as expected', () => {
    const gen = saga.twitterLoginSaga();

    gen.next();
    gen.throw('error');
    gen.next();
    expect(gen.next().done).toBeTruthy();
});

it('Should have a watcher', () => {
    const gen = TwitterLoginSaga();

    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
})
