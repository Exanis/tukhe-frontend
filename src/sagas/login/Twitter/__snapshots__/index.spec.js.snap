// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should execute steps as expected when there is a token 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should execute steps as expected when there is a token 2`] = `Promise {}`;

exports[`Should execute steps as expected when there is a token 3`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [
        500,
      ],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should execute steps as expected when there is a token 4`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should execute steps as expected when there is a token 5`] = `Promise {}`;

exports[`Should execute steps as expected when there is a token 6`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": "test",
        "type": "user/setToken",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;

exports[`Should execute steps as expected when there is no token 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should execute steps as expected when there is no token 2`] = `Promise {}`;

exports[`Should execute steps as expected when there is no token 3`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [
        500,
      ],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should execute steps as expected when there is no token 4`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should execute steps as expected when there is no token 5`] = `Promise {}`;

exports[`Should have a watcher 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [
        [Function],
        [Function],
      ],
      "context": null,
      "fn": [Function],
    },
    "type": "FORK",
  },
}
`;
